spec/models/rbac_spec.rb

```diff
@@ -17,6 +17,58 @@
     @user  = FactoryGirl.create(:user, :miq_groups => [@group])
   end
 
+  context "tenant scoping" do
+    before do
+      default_tenant = Tenant.seed
+
+      @owner_tenant  = FactoryGirl.create(:tenant, :divisible => false, :parent => default_tenant)
+      @owner_group   = FactoryGirl.create(:miq_group, :tenant => @owner_tenant)
+      @owner_user    = FactoryGirl.create(:user, :userid => 'foo', :miq_groups => [@owner_group])
+
+      @other_tenant  = FactoryGirl.create(:tenant, :divisible => false, :parent => default_tenant)
+      @other_group   = FactoryGirl.create(:miq_group, :tenant => @other_tenant)
+      @other_user    = FactoryGirl.create(:user, :userid => 'bar', :miq_groups => [@other_group])
+
+      @owner_vm      = FactoryGirl.create(:vm_vmware, :tenant => @owner_tenant)
+    end
+
+    it ".search with :userid, finds user's tenant vms" do
+      results, = Rbac.search(:class => "Vm", :results_format => :objects, :userid => @owner_user.userid)
+      expect(results).to eq [@owner_vm]
+    end
+
+
+    it ".search with :userid filters out other tenants" do
+      results, = Rbac.search(:class => "Vm", :results_format => :objects, :userid => @other_user.userid)
+      expect(results).to eq []
+    end
+
+    it ".search with User.with_userid finds user's tenant vms" do
+      User.with_userid(@owner_user.userid) do
+        results, = Rbac.search(:class => "Vm", :results_format => :objects)
+        expect(results).to eq [@owner_vm]
+      end
+    end
+
+    it ".search with User.with_userid filters out other tenants" do
+      User.with_userid(@other_user.userid) do
```

Comment:
> Would prefer to remove User.with_user calls. But not sure if that is reasonable.
But at least lets keep that out of RBAC if possible, have user as a parameter coming in.

We need @dclarizio's insights, I don't understand the use cases of `User.with_user` with a block vs. passing in `:userid` or `:miq_group_id` to Rbac.search.  @gtanzillo and I believed we might have a situation where the current user might want to run a search as a different user.  It seems reasonable to do.

I guess the question is do we need to support `User.with_user` AND `Rbac.search(:userid => ...) AND `Rbac.search(:miq_group_id => ...)` for tenancy... This PR allows for each but I made no changes or assertions about providing more than one and definitely didn't touch when they conflict.